src/lib/TypedEmitter.ts

```diff
@@ -0,0 +1,45 @@
+import EventEmitter from 'events'
+import { applyMixin } from './shared'
+
+export type EventHandler =
```

Comment:
Why do we need to have these types? Shouldn't there be typings available for EventEmitter already?

Reply:
A friend of mine (@AlCalzone, a TS wizard 🧙🏼‍♂️ ) suggested me this a while ago. Based on what he told me the `TypedEventEmitter` is far better because:

> you get autocomplete for the events that exist, and you get syntax help for the callback parameters, and typescript will yell at you if you accidentally use an event wrong. No more typos in event names or non-existent events, no more wrong callback params, no more looking up elsewhere what callback params there are

Reply:
Examples for a class that is such a `TypedEventEmitter`:
![grafik](https://github.com/mqttjs/MQTT.js/assets/17641229/d80ea625-1533-4967-a145-938b636c42e7)
![grafik](https://github.com/mqttjs/MQTT.js/assets/17641229/f25543d7-67f6-45b8-8fb1-1836d8f143c8)
![grafik](https://github.com/mqttjs/MQTT.js/assets/17641229/080939cb-6a2e-435c-8919-6a8035a1c2d8)

Reply:
Ah, neat!